gio/src/initable.rs

```diff
@@ -18,6 +23,21 @@ impl Initable {
         unsafe { object.init(cancellable)? };
         Ok(object)
     }
+
+    // rustdoc-stripper-ignore-next
+    /// Create a new instance of an initable object of the given type with the given properties.
+    ///
+    /// Similar to [`Object::with_type`] but can fail either because the object
+    /// creation failed or because `Initable::init` failed.
+    pub fn with_type(
+        type_: Type,
+        properties: &[(&str, &dyn ToValue)],
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<Object, InitableError> {
+        let object = Object::with_type(type_, properties)?;
```

Comment:
You should probably add an assertion here that `type_` is a `Initiable`?

Reply:
Shall I assert of should I return `NewObjectFailed`? I guess the latter, since also `Object::with_type` errors out rather than asserting

Reply:
Return an error, yes